src/main/kotlin/app/utils/logging/LoggerUtils.kt

```diff
@@ -0,0 +1,157 @@
+package app.utils.logging
+
+/**
+This file had simple json logging utils extending SLF4J in a standard way.
+It brings all our changes into one code module and one test file, and allows us to make a simple override in the
+resources/logback.xml file so that all vagaries are in code and unit-testable; one "just" does this:
+
+<appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+<encoder class="ch.qos.logback.core.encoder.LayoutWrappingEncoder">
+<layout class="app.utils.logging.LoggerLayoutAppender" />
+</encoder>
+</appender>
+
+See http://logback.qos.ch/manual/layouts.html for examples.
+ */
+
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.classic.spi.ThrowableProxyUtil
+import ch.qos.logback.core.CoreConstants
+import ch.qos.logback.core.LayoutBase
+import org.apache.commons.text.StringEscapeUtils
+import org.slf4j.Logger
+import java.net.InetAddress
+import java.text.SimpleDateFormat
+import java.util.*
+
+
+private val defaultFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS") // 2001-07-04T12:08:56.235
+private var topic_name = System.getProperty("topic_name", "NOT_SET")
+private var hostname = InetAddress.getLocalHost().hostName
+private var environment = System.getProperty("environment", "NOT_SET")
+private var application = System.getProperty("application", "NOT_SET")
+private var app_version = System.getProperty("app_version", "NOT_SET")
+private var component = System.getProperty("component", "NOT_SET")
+private var staticData = makeLoggerStaticDataTuples()
+
+fun makeLoggerStaticDataTuples(): String {
+    return "\"topic_name\":\"$topic_name\", " +
```

Comment:
might want to consider stringbuilders if the cost of this concatenating shows up on profiling.